deviations/forms.py

```diff
@@ -51,3 +51,35 @@ def __init__(self, *args, **kwargs):
         self.fields["submitter"].widget.attrs["class"] = "search-select"
         self.fields["submitter"].help_text = ""
         self.fields["submitter"].queryset = course_instance.get_student_profiles()
+
+
+class RemoveDeadlineRuleDeviationForm(forms.Form):
+    module = forms.ModelMultipleChoiceField(
+        queryset=CourseModule.objects.none(),
+        required=False
+    )
+    exercise = forms.ModelMultipleChoiceField(
+        queryset=BaseExercise.objects.none(),
+        required=False
+    )
+    submitter = forms.ModelMultipleChoiceField(
+        queryset=UserProfile.objects.none(),
+        required=False,
```

Comment:
If the `submitter` field is not required, what happens when it is left empty? Are all deviations removed for the given exercises or is nothing removed?

Reply:
Those fields are not required in the add view either. Leaving submitter empty removes nothing. This could be changed in adding of deviations, too.

Reply:
Submitting empty values does not make sense so it should not validate. The user should press the cancel button if they want to cancel everything. Does the form have a cancel button?

The form validation can be changed so that it requires submitters and either modules or exercises. The form class has a method that is very suitable for validating data that depends on multiple fields ("either module or exercise must be set"). That is the `clean` method.

https://docs.djangoproject.com/en/2.2/ref/forms/validation/